crates/trigger-http/src/lib.rs

```diff
@@ -715,7 +723,9 @@ impl OutboundWasiHttpHandler for HttpRuntimeData {
                 "self".into()
             };
             eprintln!("To allow requests, add 'allowed_outbound_hosts = [\"{}\"]' to the manifest component section.", host);
-            anyhow::bail!("destination-not-allowed (error 1)")
+            return Err(HttpError::trap(anyhow::anyhow!(
```

Comment:
I realize we were trapping before, but I do wonder if we should be returning `HTTP-request-denied` instead.